server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -144,30 +143,24 @@ public MapperService(
         this.indexVersionCreated = indexSettings.getIndexVersionCreated();
         this.indexAnalyzers = indexAnalyzers;
         this.mapperRegistry = mapperRegistry;
-        Function<DateFormatter, MappingParserContext> parserContextFunction = dateFormatter -> new MappingParserContext(
+        this.mappingParserContext = new MappingParserContext(
```

Comment:
The only real change here is that there's now one context instance for the lifetime of each mapper service, rather than a new one created on each merge.  I don't think there will be any measurable effect on memory or cpu.